_specifications/specification-3-16.md

```diff
@@ -1591,6 +1591,28 @@ interface ClientCapabilities {
 		*/
 		workspaceFolders?: boolean;
 
+		/**
+		* The client has support for rename requests/notifications.
+		*
+		* Since 3.16.0
+		*/
+		renameFiles?: {
+			/**
+			 * The client has support for didRenameFiles notifications.

```

Comment:
Maybe something like: "The client supports sending didRenameFiles notifications." so it's obvious which direction the communication flows?

Reply:
Assuming that we will add support for create and delete as well (@DanTup would you be willing to do so) to make this a complete story we can either do

```typescript
files: {
   willCreate: boolean;
   willCreateWaitUntil: boolean;
   didCreate: boolean;
   willRename: boolean;
   willRename: boolean;
   willRenameWaitUntil: boolean;
   willDelete: boolean;
   willDeleteWaitUntil: boolean;
   didDelete: boolean;
}
```

or we could do 

```typescript
files: {
    create: {
        ...
    },
    rename: {
        ....
    },
    delete: {
    }
}
```

I am actually for the first option.